packages/ra-core/src/core/CoreAdminContext.tsx

```diff
@@ -73,9 +73,13 @@ React-admin requires a valid dataProvider function to work.`);
             <AuthContext.Provider value={finalAuthProvider}>
                 <DataProviderContext.Provider value={finalDataProvider}>
                     <TranslationProvider i18nProvider={i18nProvider}>
-                        <ConnectedRouter history={finalHistory}>
-                            {children}
-                        </ConnectedRouter>
+                        {typeof window !== 'undefined' ? (
+                            <ConnectedRouter history={finalHistory}>
+                                {children}
+                            </ConnectedRouter>
+                        ) : (
+                            children
```

Comment:
I don't understand: calls to `<Route>` in react-admin components (e.g. `<Resource>`) will fail if there is no router. This can only work if developers wrap their entire Admin in a StaticRouter. We should detect that a router is already provided and throw a nice exception if there is none, otherwise the developer experience will be confusing.  

Or am I missing something?

Reply:
Hmm... `react-router` will throw an error if `Route` is used outside of `Router`
https://github.com/ReactTraining/react-router/blob/master/packages/react-router/modules/Route.js#L35

I believe this is enough and that it shouldn't be the responsibility of `AdminContext` to enforce that.
BUT say the word and I'll add a check and a different(?) error message.

Reply:
> This can only work if developers wrap their entire Admin in a StaticRouter.

And yes, all SSR frameworks do this (or require the dev to do it) as it requires tight integration with the server's request.